src/components/20-molecules/datepicker/index.js

```diff
@@ -0,0 +1,369 @@
+            <div class="m-datepicker__input-wrap">
+              <input
+                @keyup="${this.handleInputChange}"
+                class="m-datepicker__input js-datepicker__input"
+                type="text"
+                placeholder="Please select a date"
+                value="${this.outputDate}"
+              />
+              <button class="m-datepicker__input-button" @click="${this.handleInputButtonClick}">
+                <span class="m-datepicker__input-icon">${iconDatepicker}</span>
+              </button>
+            </div>
+          `}
+        <div class="m-datepicker__wrap">
+          <div class="m-datepicker__article">
+            <div class="m-datepicker__dropdown-wrap">
+              <axa-dropdown
+                class="m-datepicker__dropdown m-datepicker__dropdown-month js-datepicker__dropdown-month"
+                max-height
+                items="${JSON.stringify(this.monthItems)}"
+                title="Month"
+              >
+              </axa-dropdown>
+
+              <axa-dropdown
+                class="m-datepicker__dropdown m-datepicker__dropdown-year js-datepicker__dropdown-year"
+                max-height
+                items="${JSON.stringify(this.yearItems)}"
+                title="Year"
+              >
+              </axa-dropdown>
+            </div>
+
+            <div class="m-datepicker__weekdays">
+              ${this.weekdays &&
+                this.weekdays.map(
+                  day =>
+                    html`
+                      <div class="m-datepicker__weekdays-day">${day}</div>
+                    `
+                )}
+            </div>
+
+            <div class="m-datepicker__calendar js-datepicker__calendar">
+              ${this.cells &&
+                this.cells.map(
+                  (cell, index) =>
+                    html`
+                      <button
+                        @click="${this.handleDatepickerCalendarCellClick}"
+                        tabindex="0"
+                        data-index="${index}"
+                        data-value="${cell.value}"
+                        class="m-datepicker__calendar-cell ${cell.className}"
+                      >
+                        ${cell.text}
+                      </button>
+                    `
+                )}
+            </div>
+            <div class="m-datepicker__buttons">
+              <axa-button
+                ghost
+                class="m-datepicker__button m-datepicker__button-cancel js-datepicker__button-cancel"
+                @click="${this.handleButtonCancelClick}"
+                >${this.labelButtonCancel}</axa-button
+              >
+              <axa-button class="m-datepicker__button m-datepicker__button-ok js-datepicker__button-ok" @click="${this.handleButtonOkClick}"
+                >${this.labelButtonOk}</axa-button
+              >
+            </div>
+          </div>
+        </div>
+      </article>
+    `;
+  }
+
+  // Events
```

Comment:
i like this! all events in one place! nice and clean. should become best practice and added to our contribution.md file

Reply:
😘